src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -120,6 +120,16 @@ public function load(array $configs, ContainerBuilder $container): void
         $patchFormats = $this->getFormats($config['patch_formats']);
         $errorFormats = $this->getFormats($config['error_formats']);
         $docsFormats = $this->getFormats($config['docs_formats']);
+        $jsonSchemaFormats = $config['jsonschema_formats'];
+
+        if (!$jsonSchemaFormats) {
+            foreach (array_keys($formats) as $f) {
+                // Distinct JSON-based formats must have names that start with 'json'
+                if (str_starts_with($f, 'json') && 'json' !== $f) {
```

Comment:
```suggestion
                if (str_starts_with($f, 'json')) {
```

we can keep standard json no?

Reply:
Yes we do. The goal was to work with the behaviour of `distinctFormats` here:
https://github.com/api-platform/core/blob/4f93952f9201a46416f8d5b21eae440553a86db3/src/JsonSchema/SchemaFactory.php#L259-L262

I'll just shift the logic into the factory, ergo:
```
if ($format !== 'json' && $this->distinctFormats[$format] ?? false) {
```

Otherwise we end up with a bunch of `ResourceName.json` schema components.